core/main.go

```diff
@@ -72,59 +100,70 @@ func main() {
 				RootCAs:            caCertPool,
 				InsecureSkipVerify: *insecureSkipVerify,
 			},
+			TLSHandshakeTimeout: 5 * time.Second,
 		}
 	}
 
-	client := &http.Client{Timeout: 10 * time.Second, Transport: tr}
+	// 2/2 Build HTTP Client
+	client := &http.Client{Transport: tr, Timeout: 10 * time.Second}
 
-	a, err := api.NewClient(&api.Config{
-		Address:    secretServiceBaseURL,
-		HttpClient: client,
-	})
-	s := a.Sys()
-	inited := false
-	sealed := true
+	// Loop duration interval between Vault init and unseal retries
+	intervalDuration := time.Duration(*waitInterval) * time.Second
+	// Loop exit condition
+	loopExit := false
```

Comment:
since there's no nested for loops here, you could remove this variable and instead do `break` everywhere you set `loopExit = true`